tool_size: [0.16, 0.01, 0.02]